ProductTree/ViewModels/ProductTreeViewModel.cs

```diff
@@ -150,13 +150,17 @@ public ProductTreeViewModel(Option option, ISession session, IThingDialogNavigat
                 .Select(x => x.ChangedThing as Iteration)
                 .ObserveOn(RxApp.MainThreadScheduler)
                 .Subscribe(this.SetTopElement);
-            this.Disposables.Add(iterationSubscription);
-
-            this.AddSubscriptions();
 
-            this.SetTopElement(iteration);
+            this.Disposables.Add(iterationSubscription);
 
-            this.UpdateProperties();
+            this.ExecuteLongRunningDispatcherAction(
+                () => 
+                { 
+                    this.AddSubscriptions();
+                    this.SetTopElement(iteration);
+                    this.UpdateProperties();
+                },
+                "Loading browser...");
```

Comment:
i propose we make this specific to the browser, please rephrase to `Loading Product Tree`. I think the `PanelCaption` can be used here